packages/node/src/transports/http.ts

```diff
@@ -100,6 +104,13 @@ export class HTTPTransport implements Transport {
           resolve(mapHttpMessageToResponse(res));
         });
       });
+      // set timeout within promise so that it can resolve itself if time is exceeded
+      if (limitInMs > 0) {
+        timeoutId = setTimeout(() => {
+          req.destroy();
+          resolve({ status: Status.Timeout, statusCode: 0 });
```

Comment:
small ask - since we now have more control of the response, maybe we could include a `Body` that includes metadata on how long the request waited before timing out in ms, or when the network request started? Non-blocking but would prefer to have a TODO/follow-up if not addressed currently.

This might also help our users debug what exactly is going on,

Reply:
I'll make a ticket to follow up on this